include/deal.II/lac/linear_operator.h

```diff
@@ -0,0 +1,632 @@
+  {
+    first_op.vmult_add(v, u);
+    second_op.vmult_add(v, u);
+  };
+
+  return_op.Tvmult = [first_op, second_op](Domain &v, const Range &u)
+  {
+    first_op.Tvmult(v, u);
+    second_op.Tvmult_add(v, u);
+  };
+
+  return_op.Tvmult_add = [first_op, second_op](Domain &v, const Range &u)
+  {
+    first_op.Tvmult_add(v, u);
+    second_op.Tvmult_add(v, u);
+  };
+
+  return return_op;
+}
+
+
+/**
+ * Subtraction of two linear operators @p first_op and @p second_op given
+ * by $(\text{first_op}-\text{second_op})x:=\text{first_op}(x)-\text{second_op}(x)$
+ */
+template <typename Range, typename Domain>
+LinearOperator<Range, Domain>
+operator-(const LinearOperator<Range, Domain> &first_op,
+          const LinearOperator<Range, Domain> &second_op)
+{
+  // implement with addition and scalar multiplication
+  return first_op + (-1. * second_op);
+}
+
+
+/**
+ * Concatenation of two linear operators @p first_op and @p second_op given
+ * by $(\text{first_op}*\text{second_op})x:=\text{first_op}(\text{second_op}(x))$
+ */
+template <typename Range, typename Intermediate, typename Domain>
+LinearOperator<Range, Domain>
+operator*(const LinearOperator<Range, Intermediate> &first_op,
+          const LinearOperator<Intermediate, Domain> &second_op)
+{
+  LinearOperator<Range, Domain> return_op;
+
+  return_op.new_domain_member = second_op.new_domain_member;
+  return_op.new_range_member = first_op.new_domain_member;
+
+  // ensure to have valid computation objects by catching first_op and
+  // second_op by value
+
+  return_op.vmult = [first_op, second_op](Range &v, const Domain &u)
+  {
+    // FIXME: Intermediate storage
+    Intermediate i = second_op.new_range_member();
+    second_op.vmult(i,u);
+    first_op.vmult(v,i);
+  };
+
+  return_op.vmult_add = [first_op, second_op](Range &v, const Domain &u)
+  {
+    // FIXME: Intermediate storage
+    Intermediate i = second_op.new_range_member();
+    second_op.vmult(i,u);
+    first_op.vmult_add(v,i);
+  };
+
+  return_op.Tvmult = [first_op, second_op](Domain &v, const Range &u)
+  {
+    // FIXME: Intermediate storage
+    Intermediate i = first_op.new_domain_member();
+    first_op.Tvmult(i,u);
+    second_op.Tvmult(v,i);
+  };
+
+  return_op.Tvmult_add = [first_op, second_op](Domain &v, const Range &u)
+  {
+    // FIXME: Intermediate storage
+    Intermediate i = first_op.new_domain_member();
+    first_op.Tvmult(i,u);
+    second_op.Tvmult_add(v,i);
+  };
+
+  return return_op;
+}
+
+
+/**
+ * Scalar multiplication of a ScalarOperator object from the left.
+ *
+ * The Domain and Range types must implement the following
+ * <code>operator*=</code> member functions for rescaling:
+ *
+ * @code
+ * Domain & operator *=(Number);
+ * Range & operator *=(Number);
+ * @endcode
+ */
+template <typename Number, typename Range, typename Domain>
+LinearOperator<Range, Domain>
+operator*(Number number,
+          const LinearOperator<Range, Domain> &op)
```

Comment:
You probably also want the operator with reversed arguments. By the way, `Number` matches a whole lot of things here, not just scalars. Since you require C++11 anyway, want to do something like
```
  static_assert (...check whether Number is a scalar...)
```
? Or maybe use
```
template <typename Range, typename Domain>
...
operator * (typename Range::value_type number,
                const LinearOperator<R,D> &op)
```
?